src/pretix/api/serializers/orderchange.py

```diff
@@ -0,0 +1,423 @@
+# applicable granting you additional permissions and placing additional restrictions on your usage of this software.
+# Please refer to the pretix LICENSE file to obtain the full terms applicable to this work. If you did not receive
+# this file, see <https://pretix.eu/about/en/license>.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied
+# warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Affero General Public License for more
+# details.
+#
+# You should have received a copy of the GNU Affero General Public License along with this program.  If not, see
+# <https://www.gnu.org/licenses/>.
+#
+import logging
+import os
+
+import pycountry
+from django.core.files import File
+from rest_framework import serializers
+from rest_framework.exceptions import ValidationError
+
+from pretix.api.serializers.order import (
+    AnswerCreateSerializer, AnswerSerializer, CompatibleCountryField,
+    OrderPositionCreateSerializer,
+)
+from pretix.base.models import ItemVariation, Order, OrderFee, OrderPosition
+from pretix.base.services.orders import OrderError
+from pretix.base.settings import COUNTRIES_WITH_STATE_IN_ADDRESS
+
+logger = logging.getLogger(__name__)
+
+
+class OrderPositionCreateForExistingOrderSerializer(OrderPositionCreateSerializer):
+    order = serializers.SlugRelatedField(slug_field='code', queryset=Order.objects.none(), required=True, allow_null=False)
+    answers = AnswerCreateSerializer(many=True, required=False)
+    addon_to = serializers.IntegerField(required=False, allow_null=True)
+    secret = serializers.CharField(required=False)
+    attendee_name = serializers.CharField(required=False, allow_null=True)
+    seat = serializers.CharField(required=False, allow_null=True)
+    price = serializers.DecimalField(required=False, allow_null=True, decimal_places=2,
+                                     max_digits=10)
+    country = CompatibleCountryField(source='*')
+
+    class Meta:
+        model = OrderPosition
+        fields = ('order', 'item', 'variation', 'price', 'attendee_name', 'attendee_name_parts', 'attendee_email',
+                  'company', 'street', 'zipcode', 'city', 'country', 'state',
+                  'secret', 'addon_to', 'subevent', 'answers', 'seat')
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        if not self.context:
+            return
+        self.fields['order'].queryset = self.context['event'].orders.all()
+        self.fields['item'].queryset = self.context['event'].items.all()
+        self.fields['subevent'].queryset = self.context['event'].subevents.all()
+        self.fields['seat'].queryset = self.context['event'].seats.all()
+        self.fields['variation'].queryset = ItemVariation.objects.filter(item__event=self.context['event'])
+        if 'order' in self.context:
+            del self.fields['order']
+
+    def validate(self, data):
```

Comment:
Is this missing a `super().validate(data)`?

Reply:
Hmm, I think that the parent function doesn't do anything, but it is more correct to still call it, will change it.